src/coprocessor/statistics/analyze.rs

```diff
@@ -77,23 +81,44 @@ impl<S: Snapshot> AnalyzeContext<S> {
     // it would build a histogram and count-min sketch of index values.
     fn handle_index(
         req: AnalyzeIndexReq,
-        scanner: &mut IndexScanExecutor<TiKVStorage<SnapshotStore<S>>>,
+        scanner: &mut RangesScanner<TiKVStorage<SnapshotStore<S>>>,
+        is_common_handle: bool,
     ) -> Result<Vec<u8>> {
         let mut hist = Histogram::new(req.get_bucket_size() as usize);
         let mut cms = CmSketch::new(
             req.get_cmsketch_depth() as usize,
             req.get_cmsketch_width() as usize,
         );
-        while let Some(row) = scanner.next()? {
-            let row = row.take_origin()?;
-            let (bytes, end_offsets) = row.data.get_column_values_and_end_offsets();
-            hist.append(bytes);
-            if let Some(c) = cms.as_mut() {
-                for end_offset in end_offsets {
-                    c.insert(&bytes[..end_offset])
+
+        while let Some((key, _)) = scanner.next()? {
+            let mut key = &key[..];
+            if is_common_handle {
+                table::check_record_key(key)?;
+                key = &key[table::PREFIX_LEN..];
```

Comment:
Please confirm whether it is the correct place to handle record in `handle_index` function

Reply:
Here is the handler of `analyzeIndexCommonHandle` in `unistore` which cut the row key prefix first then decode the datum: https://github.com/pingcap/tidb/blob/4d9428677c7c7a39c4661292f585003bbfbf54c2/store/mockstore/unistore/cophandler/analyze.go#L162

Reply:
I mean this function is called `handle_index` but it is processing table records and I'm not sure whether this is correct. The code snippet does not have such problems. It's called `analyzeIndexCommonHandle` and does exactly as its name.